0017-negotiation-and-execution-protocol.adoc

```diff
@@ -0,0 +1,451 @@
+= Separating negotiation and execution phase =
+Thomas Eizinger <thomas@coblox.tech>; Philipp Hoenisch <philipp@coblox.tech>; Daniel Karzel <daniel.karzel@coblox.tech>;
+:toc:
+:revdate: 2019-10-05
+:sectnums:
+:sectnumlevels: 5
+
+NOTE: Author: {authors} +
+Date: {revdate} +
+Tracking issue: https://github.com/comit-network/comit-rs/issues/1413[#1413]
+
+== Context ==
+
+Over time we came to the conclusion that our protocol stack in general, and RFC003 in particular, is difficult to understand in certain ways.
+
+We could specify several problems:
+
+. The Accept/Decline _feels weird_. It often does not make sense to _accept_ a trade which was already agreed on before (e.g. on application layer).
+. We do not clearly distinguish rate negotiation from the execution at the moment.
+. It is hard to add the concept of orders and an orderbook to COMIT.
+
+=== This spike includes
+
+. Solution proposal for splitting trade negotiation and execution in different protocols.
+. Solution proposal for the "How to update an order"/"Orders should not get stale" problem.
+. Defining the base for a first negotiation protocol.
+. Defining the changes needed to the existing execution protocol RFC003.
+
+=== This spike does not include
+
+The needed API changes to facilitate the newly proposed execution protocol.
+
+== TL;DR & Next step proposal
+
+This spike proposes a new protocol to exchange trade orders between a maker and a taker including an update to the execution protocol as defined in RFC003.
+
+We propose the following steps to achieve early validations of the proposed protocol (note: this proposal assumes that none of the assumptions have been refuted and does not make any assumptions on the size of the tickets or PRs):
+
+. Integrate into the https://github.com/comit-network/create-comit-app/[create-commit-app] a simple request/response example to exchange of order parameters (KISS: keep it simple stupid).
+. Use the output of previous step and send a _swap request_ from taker to maker.
+. Automate _accept swap request_ based on the defined parameters in the first few steps.
+. Integrate the proposed protocol including auto-accept into the https://github.com/comit-network/comit-js-sdk[comit-js-sdk].
+. Move _auto-accept_ into cnd and change cnd's API to reflect the _feeling_ of the proposed new <<execution-protocol, execution protocol>> (no actual changes of the execution protocol are needed at this point).
+. Update the proposed negotiation protocol to libp2p if needed.
+. Update the communication between two cnd nodes to reflect the proposed execution protocol.
```

Comment:
When do we review the execution protocol?

Reply:
If you have any comments, feel free to provide them but I would expect another more clearly defined document for the execution protocol changes.